src/main/java/nextstep/qna/domain/Answer.java

```diff
@@ -76,4 +77,15 @@ public void toQuestion(Question question) {
     public String toString() {
         return "Answer [id=" + getId() + ", writer=" + writer + ", contents=" + contents + "]";
     }
+
+    public void validate(NsUser nsUser) throws CannotDeleteException {
+        if (!writer.matchUser(nsUser)) {
+            throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
+        }
+    }
+
+    public DeleteHistory delete() {
+        this.deleted = true;
+        return DeleteHistory.answerOf(id, writer);
+    }
```

Comment:
```suggestion
    public DeleteHistory delete(NsUser nsUser) {
        validate(nsUser);
        this.deleted = true;
        return DeleteHistory.answerOf(id, writer);
    }
```

삭제할 때 검증하는 방식은 어떨까요?

Reply:
만약 
질문(작성자 : 유저1) 의 답변1(작성자: 유저1), 답변2(작성자 : 유저2)가 있다라고 
가정 했을때 답변1은 삭제 시 검증 로직이 통과 되어 deleted = true로 바뀝니다

그리고 답변2의 삭제 메소드 내부에 검증을 하면 예외 발생으로 답변1 deleted의 상태를 다시 변경 해야 되는 거 아닌가요???

Reply:
트랜잭션을 하나로 묶었다면 롤백되어 문제가 없을것 같아요